integration-tests/tests/client/process_blocks.rs

```diff
@@ -1558,6 +1575,38 @@ fn test_gc_after_state_sync() {
     assert!(env.clients[1].chain.clear_data(tries, 2).is_ok());
 }
 
+#[test]
+fn test_num_blocks_in_storage_config_setting() {
+    let epoch_length = 128;
+    // TODO - figured out how to change `num_epochs_to_keep_store_data = 10`
+    let mut genesis = Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
+    genesis.config.epoch_length = epoch_length;
+    let mut chain_genesis = ChainGenesis::test();
+    chain_genesis.epoch_length = epoch_length;
+    let num_epochs_to_keep_store_data = 10;
+    let mut env = TestEnv::builder(chain_genesis)
+        .clients_count(2)
+        .runtime_adapters(create_nightshade_runtimes_with_num_epochs(
+            &genesis,
+            2,
+            Some(num_epochs_to_keep_store_data),
+        ))
+        .build();
+    for i in 1..epoch_length * (num_epochs_to_keep_store_data) + 2 {
+        let block = env.clients[0].produce_block(i).unwrap().unwrap();
+        env.process_block(0, block.clone(), Provenance::PRODUCED);
+        env.process_block(1, block, Provenance::NONE);
+    }
+    // Check for GC'ed blocks.
```

Comment:
I'm confused why the number is 100 here. Shouldn't it be a number close to 128?

Reply:
@mzhangmzz I'm confused as well. Blocks from 1..100 are unavailable, wile blocks 101... are available to read.

Reply:
@mzhangmzz I added comments to the test.

```

/// This field tests `num_epochs_to_keep_store_data` setting.
/// We test, we keep the number of epochs proportional to the data.
///
/// For this test, we set epoch length to `128`, and `num_epochs_to_keep_store_data` to 10.
/// This tests shows, we keep 9 full epochs, and one partial epoch with `128-100` blocks.
/// I know it's weird, right?
```